src/libYARP_serversql/src/yarpserver.cpp

```diff
@@ -224,62 +213,56 @@ yarpserversql_API int yarpserver_main(int argc, char *argv[]) {
         printf("  --no-web-cache           Reload pages from file for each request.\n");
         printf("  --ros                    Delegate pub/sub to ROS name server.\n");
         printf("  --silent                 Start in silent mode.\n");
-        return 0;
-    } else {
-        fprintf(out, "Call with --help for information on available options\n");
+        this->stop();
```

Comment:
The `stop` method of the Thread should not be called by the thread itself, see https://github.com/robotology/yarp/blob/e0ca8e6f73c37286734d06e3afb8015529ee36b9/src/libYARP_OS/include/yarp/os/Thread.h#L76 . Returning from the run should be sufficient.